### PR TITLE
Server programs: try to print error message if can't connect to DB

### PR DESCRIPTION
**Pull request by davidpanderson**

### CODE REVIEW

File: db/db_base.cpp [hunk 1]
```diff
@@ -85,10 +85,14 @@ int DB_CONN::open(
     // CLIENT_FOUND_ROWS means that the # of affected rows for an update
     // is the # matched by the where, rather than the # actually changed
     //
-    mysql = mysql_real_connect(
+    MYSQL* mysql2 = mysql_real_connect(
```
Comment (AenBleidd) [inline db/db_base.cpp:88]: What is the reason of adding new variable here? In case of unsuccessful connect, `mysql` variable will still be uninitialized.

Comment (davidpanderson) [reply db/db_base.cpp:88]: If the connect failed, mysql will be NULL.  mysql_errno() and mysql_error() require a non-NULL argument.

Comment (AenBleidd) [reply db/db_base.cpp:88]: But will `mysql` will contain relevant error information then?

Comment (davidpanderson) [reply db/db_base.cpp:88]: Yes.  It will contain info about the fail connect.

**Pull request merged**